### PR TITLE
Feature/booking model se 1016

### PR DESCRIPTION
**Pull request by peteryates**
### Context

Remove the hardcoded bookings data from the bookings controllers and replace with an actual model

### Changes proposed in this pull request

Implement the `Bookings::Booking` model in a basic form. It has relationships with the school, the selected subject and the placement request from which it will be generated, plus it holds the finalised date.

### Guidance to review

Ensure the model has the required attributes and the relationships are correct.

### CODE REVIEW

File: app/models/bookings/booking.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+module Bookings
+  class Booking < ApplicationRecord
+    belongs_to :bookings_placement_request,
+      class_name: 'Bookings::PlacementRequest',
+      dependent: :destroy
```
Comment (rjlynch) [inline app/models/bookings/booking.rb]: Not sure if we want this dependent destroy. Probably want to keep the placement request around if the school deletes the booking

File: app/models/bookings/booking.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    belongs_to :bookings_subject,
+      class_name: 'Bookings::Subject',
+      dependent: :destroy
+
+    belongs_to :bookings_school,
+      class_name: 'Bookings::School',
+      dependent: :destroy
```
Comment (rjlynch) [inline app/models/bookings/booking.rb]: Probably don't want this dependent destroy either

File: app/models/bookings/booking.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+module Bookings
+  class Booking < ApplicationRecord
+    belongs_to :bookings_placement_request,
+      class_name: 'Bookings::PlacementRequest',
+      dependent: :destroy
+
+    belongs_to :bookings_subject,
+      class_name: 'Bookings::Subject',
+      dependent: :destroy
```
Comment (rjlynch) [inline app/models/bookings/booking.rb]: Another one here

File: app/models/bookings/booking.rb (hunk 1 above)
Comment (peteryates) [reply app/models/bookings/booking.rb]: Wouldn't this mean that if the placement request was deleted the booking would be too?

**Pull request merged**